File: 5_Fine_Grained_Parallelism/OpenMP/Matrix.cpp
#include "stdafx.h"
#include <iostream>
#include <cstdlib>
#include "Data.h"

using namespace std;

Matrix::Matrix(int size, int number)
{
	_size = size;
	_data = new int*[size];
	for (int i = 0; i < size; i++)
		_data[i] = new int[size];

	for (int i = 0; i < size; i++)
		for (int j = 0; j < size; j++)
			_data[i][j] = number;
}

Matrix::Matrix(int size)
{
	_size = size;
	_data = new int*[size];
	for (int i = 0; i < size; i++)
		_data[i] = new int[size];
}

Matrix::Matrix(Matrix* matrix)
{
	_size = matrix->getSize();
	_data = new int*[_size];
	for (int i = 0; i < _size; i++)
		_data[i] = new int[_size];

	for (int i = 0; i < _size; i++)
		for (int j = 0; j < _size; j++)
			_data[i][j] = matrix->getData()[i][j];
}

Matrix::~Matrix()
{
	for (int i = 0; i < _size; i++)
		delete[] _data[i];
	delete[] _data;
}

Matrix* Matrix::copyTo()
{
	Matrix* result = new Matrix(_size, false);
	for (int i = 0; i < _size; i++)
		for (int j = 0; j < _size; j++)
			result->setElement(i, j, _data[i][j]);
	return result;
}

void Matrix::setElement(int indexI, int indexJ, int value)
{
	this->_data[indexI][indexJ] = value;
}

int Matrix::getElement(int indexI, int indexJ)
{
	return this->_data[indexI][indexJ];
}

int Matrix::getSize()
{
	return _size;
}

int** Matrix::getData()
{
	return _data;
}

void Matrix::print()
{
	if (_size < 0)
		for (int i = 0; i < _size; i++)
		{
			for (int j = 0; j < _size; j++)
				printf("%6d ", _data[i][j]);
			printf("\n");
		}
	else
		printf("Output is too big!\n");
}

File: 6_Fork_Join/Java/ForkJoinSort/src/com/kirintor/Matrix.java
package com.kirintor;

/**
 * Created by kirintor830 on 14.09.2017.
 */

public class Matrix {

    private int n;
    private int[][] data;

    public Matrix(int n, int number){
        data = new int[n][n];
        this.n = n;
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
            {
                data[i][j] = number;
            }
    }

    public Matrix(Matrix MA){
        data = new int[MA.getN()][MA.getN()];
        this.n = MA.getN();
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
            {
                data[i][j] = MA.getElement(i, j);
            }
    }

    public Matrix(int n){
        data = new int[n][n];
        this.n = n;
    }

    public void print(){
        if (n <= 20)
        {
            for (int i = 0; i < n; i++)
            {
                System.out.print("[");
                for (int j = 0; j < n - 1; j++)
                {
                    System.out.print(data[i][j] + ", ");
                }
                System.out.print(data[i][n - 1] + "]\n");
            }
        }
        else
        {
            System.out.println("Output is too big!");
        }
    }

    public Matrix Copy(){
        Matrix result = new Matrix(n,0);
        for (int i = 0; i < this.n; i++)
            for (int j = 0; j < this.n; j++)
            {
                result.data[i][j] = data[i][j];
            }
        return result;
    }

    public int getN() {
        return n;
    }

    public void setN(int n) {
        this.n = n;
    }

    public int[][] getData()
    {
        return this.data;
    }

    public int getElement(int indexI, int indexJ) {
        return data[indexI][indexJ];
    }

    public void setElement(int element, int indexI, int indexJ) {
        data[indexI][indexJ] = element;
    }


}
